### PR TITLE
[Task-28]: Add a submit button that logs user input

### PR DESCRIPTION
**Pull request by cinderlee**
## Description

This PR handles adding a submit button that logs user input. At the moment, the API to submit the user input is still a WIP so the submit call instead logs the user input until the API call is ready to be integrated. 

Example of what console.log prints out:
Screen Shot 2023-09-04 at 9 05 50 PM

This PR also handles adding a settings config in eslintrc to resolve file paths when importing components/files. 

## How Has This Been Tested?

Tested running local dev of the FE only:

https://github.com/tally-team/tally-frontend/assets/24977851/fd015ead-f2fe-4b4b-a758-6d1e470bda6f

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -12,6 +12,13 @@
         "airbnb/hooks",
         "prettier"
     ],
+    "settings": {
```
Comment (cinderlee) [inline .eslintrc.json:15]: Followed this stackoverflow since I was getting unresolved path lint errors such as `Unable to resolve path to module './constants'`

https://stackoverflow.com/questions/55198502/using-eslint-with-typescript-unable-to-resolve-path-to-module

TLDR: We already have `eslint-plugin-import` package installed but we need to add the settings config for it as listed here in the doc and in the stackoverflow link: https://github.com/import-js/eslint-plugin-import#importextensions

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-import { StyleSheet, Text, View } from 'react-native';
+import { Button, StyleSheet, Text, View } from 'react-native';
 import AddItem from './components/AddItem';
 import AddPartyMember from './components/AddPartyMember';
 import Tax from './components/Tax';
 import Tip from './components/Tip';
 import { DispatchActions } from './constants';
 import { DispatchActionArgs, AppState } from './types';
 
+const APP_CONTAINER_COLOR = '#fff';
```
Comment (cinderlee) [inline src/App.tsx:10]: I had to create a const for this color weirdly since I was getting a react-native eslint error regarding 
```
Color literal: { backgroundColor: '#fff' }                      
react-native/no-color-literals
```

not sure if this should go in the constants file though... or how we want to store styles for each component in general?

File: .eslintrc.json [hunk 3]
```diff
@@ -39,6 +46,7 @@
         "radix": ["error", "as-needed"],
         "no-shadow": "warn",
         "no-restricted-globals": "warn",
-        "react/no-array-index-key": "warn"
+        "react/no-array-index-key": "warn",
+        "import/extensions": "off"
```
Comment (cinderlee) [inline .eslintrc.json:50]: I have this rule as off since when attempting to add extensions to the import file paths VSCode yells at me like so

Screen Shot 2023-09-04 at 6 16 55 PM

I can create a follow up ticket to investigate this, but I have this off atm to hopefully also unblock others.

File: src/App.tsx [hunk 4]
```diff
@@ -143,15 +147,21 @@ export default function App() {
           <Text>Tip: {state.tip.toFixed(2)}</Text>
         </>
         <Text>Subtotal: {state.total}</Text>
+        <Button
```
Comment (cinderlee) [inline src/App.tsx:150]: I didn't write a test for this component since I wasn't sure if the test should be written when we actually integrate the API? open to what others think about this

Comment (michello) [reply src/App.tsx:150]: Hm.. makes sense. Is there even a way to check what console.log it is? lol. We could have another diff that displays the text too if it's not too much work? Otherwise, I'm fine without test as well. Can you include the full payload of this state in your PR description? Just want to double check how items look like.

File: src/App.tsx (hunk 2 above)
Comment (michello) [reply src/App.tsx:10]: Having the styles in a central place like a styles folder/file makes sense. We can probably do this when we are implementing the design. :D

File: .eslintrc.json (hunk 3 above)
Comment (michello) [reply .eslintrc.json:50]: Sounds GTM. Can we include the ticket # and add it in a comment with a description here?

File: .eslintrc.json (hunk 1 above)
Comment (michello) [reply .eslintrc.json:15]: Good find! Thanks! :D

File: src/App.tsx [hunk 5]
```diff
@@ -59,6 +61,8 @@ export default function App() {
           tax: amount,
         };
       }
+      default:
```
Comment (cinderlee) [inline src/App.tsx:64]: forgot to say this came out of another lint error about missing default case in the switch statement, so i just had this return the current state if the dispatch actions doesn't match any known action (such as adding a party member etc).

File: .eslintrc.json (hunk 3 above)
Comment (cinderlee) [reply .eslintrc.json:52]: i added a todo comment about this :D issue here: https://github.com/tally-team/tally-frontend/issues/54

File: src/App.tsx (hunk 4 above)
Comment (cinderlee) [reply src/App.tsx:150]: added like a 1-liner that tests that the submit button appears in the app. also had to add another rule because it was complaining how a package should be listed under dep and not devDep. 

also updated the pr description!

**Review by michello — APPROVED** [review]
LGTM! Thanks for the changes :)
